oak_restricted_kernel_runtime/src/syscall.rs

```diff
@@ -0,0 +1,137 @@
+//
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use crate::{raw_syscall::Syscall, syscall};
+use core::ffi::{c_int, c_size_t, c_ssize_t, c_void};
+use strum::{Display, FromRepr};
+
+#[allow(clippy::upper_case_acronyms)]
+#[derive(Debug, Display, Eq, FromRepr, PartialEq)]
+#[repr(isize)]
+#[non_exhaustive]
+pub enum Errno {
+    /// Input/output error
+    EIO = -5,
+    /// Bad file descriptor
+    EBADF = -9,
+    /// Bad address
+    EFAULT = -14,
+    /// Invalid argument
+    EINVAL = -22,
+    /// Function not implemented
+    ENOSYS = -38,
+}
+
+#[no_mangle]
+pub extern "C" fn sys_read(fd: c_int, buf: *mut c_void, count: c_size_t) -> c_ssize_t {
+    unsafe { syscall!(Syscall::Read, fd, buf, count) }
+}
+
+pub fn read(fd: i32, buf: *mut u8, len: usize) -> Result<usize, Errno> {
```

Comment:
Since `read` and `write` are not exported to C, it might be better to pass in a slice and convert to a pointer and length here.

Reply:
+1

Reply:
Done.